rs/execution_environment/src/canister_manager/tests.rs

```diff
@@ -6648,3 +6659,442 @@ fn memory_usage_updates_increment_subnet_available_memory() {
+            .unwrap()
+            .system_state
+            .canister_version
+    );
+    assert_eq!(
+        third_num_changes + 1,
+        env2.get_latest_state()
+            .canister_state(&third_canister_id)
+            .unwrap()
+            .system_state
+            .get_canister_history()
+            .get_total_num_changes()
+    );
+    let history_entries: Vec<_> = env2
+        .get_latest_state()
+        .canister_state(&third_canister_id)
+        .unwrap()
+        .system_state
+        .get_canister_history()
+        .get_changes(2)
+        .cloned()
+        .collect();
+    assert_eq!(history_entries.len(), 2);
+    for entry in history_entries {
+        assert_matches!(entry.details(), CanisterChangeDetails::CanisterRename(_));
+    }
+
+    // Check that everything is the same after a checkpoint.
+    env2.checkpointed_tick();
+    let new_canister_exists = env2
+        .get_latest_state()
+        .canister_state(&new_canister_id)
+        .is_some();
+    let old_canister_exists = env2
+        .get_latest_state()
+        .canister_state(&canister_id2)
+        .is_some();
+    let third_canister_exists = env2
+        .get_latest_state()
+        .canister_state(&third_canister_id)
+        .is_some();
+    assert!(!new_canister_exists);
+    assert!(!old_canister_exists);
+    assert!(third_canister_exists);
+    assert_eq!(
+        third_version + 1,
+        env2.get_latest_state()
+            .canister_state(&third_canister_id)
+            .unwrap()
+            .system_state
+            .canister_version
+    );
+    assert_eq!(
+        third_num_changes + 1,
+        env2.get_latest_state()
+            .canister_state(&third_canister_id)
+            .unwrap()
+            .system_state
+            .get_canister_history()
+            .get_total_num_changes()
+    );
+    let history_entries: Vec<_> = env2
+        .get_latest_state()
+        .canister_state(&third_canister_id)
+        .unwrap()
+        .system_state
+        .get_canister_history()
+        .get_changes(2)
+        .cloned()
+        .collect();
+    assert_eq!(history_entries.len(), 2);
+    for entry in history_entries {
+        assert_matches!(entry.details(), CanisterChangeDetails::CanisterRename(_));
+    }
+}
+
+#[test]
+fn cannot_rename_from_ingress() {
+    let env = StateMachineBuilder::new().build();
+
+    let canister_id = env.install_canister_wat(EMPTY_WAT, vec![], None);
+
+    let new_canister_id = CanisterId::from_u64(3 * CANISTER_IDS_PER_SUBNET - 1);
+    let arguments = RenameCanisterArgs {
+        canister_id: canister_id.into(),
+        rename_to: RenameToArgs {
+            canister_id: new_canister_id.into(),
+            version: 0,
+            total_num_changes: 0,
+        },
+        sender_canister_version: 0,
+    };
+
+    let wasm_result = env.execute_ingress(IC_00, Method::RenameCanister, arguments.encode());
+
+    assert!(wasm_result.is_err());
```

Comment:
Let's also check that the error cause is as expected, i.e., that the message did not fail because of invalid input fields.